File: Data Structures & Algorithms/find-duplicate-integer/submission-0.java


class Solution {
    public int findDuplicate(int[] nums) {
        HashMap<Integer, Integer> mpp = new HashMap<>();
        
        // Populate the map with frequencies
        for (Integer num : nums) {
            mpp.put(num, mpp.getOrDefault(num, 0) + 1);
        }

        // Find the duplicate
        for (Map.Entry<Integer, Integer> m : mpp.entrySet()) {
            if (m.getValue() > 1) {
                return m.getKey(); // No need for explicit casting as we use generics
            }
        }

        return -1;
    }
}

File: Data Structures & Algorithms/reverse-integer/submission-0.py
class Solution:
    def rev(self,x):
     num=0
     while(x>0):
        num=num*10 + x%10
        x//=10
     return num   
    def reverse(self, x: int) -> int:
        if x<0:
            x=x*-1
            x=self.rev(x)
            if(x>=2**31-1):
                return 0
            return x*-1

        else:
            x=self.rev(x)
            if(x>=2**31-1):
                return 0
            return x        
        